tests/integration/search_test.t

```diff
@@ -0,0 +1,178 @@
+
+use ProductOpener::PerlStandards;
+
+use Test::More;
+use ProductOpener::APITest qw/:all/;
+use ProductOpener::Test qw/:all/;
+use LWP::Simple "get";
+use Data::Dumper;
+use Encode;
+use File::Basename "dirname";
+use JSON::PP;
+use ProductOpener::TestDefaults qw/:all/;
+use ProductOpener::Test qw/:all/;
+use Getopt::Long;
+
+my $test_name = "search_test";
+my $tests_dir = dirname(__FILE__);
+my $expected_dir = $tests_dir . "/expected_test_results/" . $test_name;
+
+my @products = (
+	{
+		code => '200000000034',
+		lang => "en",
+		product_name => "test_1",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "apple, milk, eggs, palm oil",
+		origin => "france",
+		serving_size => "10g",
+		packaging_text => "no",
+		action => "process",
+		type => "add",
+		".submit" => "submit"
+	},
+	{
+		code => '200000000037',
+		lang => "en",
+		product_name => "vegan & palm oil free",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "fruit, rice",
+		origin => "france",
+		serving_size => "10g",
+		packaging_text => "no",
+		action => "process",
+		type => "add",
+		".submit" => "submit"
+
+	},
+	{
+		code => '200000000038',
+		lang => "en",
+		product_name => "palm oil free & non vegan",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "apple, milk, eggs",
+		origin => "france",
+		serving_size => "10g",
+		packaging_text => "no",
+		action => "process",
+		type => "add",
+		".submit" => "submit"
+	},
+	{
+		code => '200000000039',
+		lang => "es",
+		product_name => "Vegan Test Snack",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "apple, water, palm oil",
+		origin => "spain",
+		serving_size => "5g",
+		packaging_text => "no",
+		action => "process",
+		categories => "snacks",
+		type => "add",
+		".submit" => "submit"
+	},
+	{
+		code => '200000000045',
+		lang => "es",
+		product_name => "Vegan Test Snack",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "apple, water",
+		origin => "France",
+		serving_size => "5g",
+		packaging_text => "no",
+		action => "process",
+		categories => "breakfast cereals",
+		type => "add",
+		".submit" => "submit"
+	}
+);
+
+remove_all_users();
+
+remove_all_products();
+
+wait_dynamic_front();
+
+my $ua = new_client();
+
+my %create_user_args = (%default_user_form, (email => 'bob@gmail.com'));
+create_user($ua, \%create_user_args);
+
+foreach my $ref (@products) {
+	create_product($ua, $ref);
```

Comment:
why don't you use a %default_product_form here so that your array above could be lighter ?

Reply:
@alexgarel : I tried to make the array lighter, it's somehow working in a weird way where it only takes the value of the default hash map for all the products.
I suspect it's a problem with the form this over-riding hashmap is passed in edit/create_product.